McCorrections/src/EventWeights_HH.cpp

```diff
@@ -94,37 +107,57 @@ ntuple::ProdSummary EventWeights_HH::GetSummaryWithWeights(const std::shared_ptr
             summary.totalShapeWeight = all_events->GetEntries();
             if(calc_withTopPt)
                 summary.totalShapeWeight_withTopPt = all_events->GetEntries();
+            if(calc_withPileUp){
+                summary.totalShapeWeight_withPileUp_Up = all_events->GetEntries();
+                summary.totalShapeWeight_withPileUp_Down = all_events->GetEntries();
+            }
         } catch(std::exception& ) {}
     }
     return summary;
 }
 
-std::vector<double> EventWeights_HH::GetTotalShapeWeights(const std::shared_ptr<TFile>& file,
-                                                          const WeightingMode& weighting_mode,
-                                                          const std::vector<NonResHH_EFT::Point>& eft_points,
-                                                          bool orthogonal)
+std::map<UncertaintyScale, std::vector<double>> EventWeights_HH::GetTotalShapeWeights(const std::shared_ptr<TFile>& file,
+                                                                  const WeightingMode& weighting_mode,
+                                                                  const std::vector<NonResHH_EFT::Point>& eft_points,
+                                                                  bool orthogonal)
 {
     static const WeightingMode shape_weights(WeightType::PileUp, WeightType::BSM_to_SM, WeightType::DY,
                                              WeightType::TTbar, WeightType::Wjets, WeightType::GenEventWeight);
+    std::vector<UncertaintyScale> unc_scales = {UncertaintyScale::Up, UncertaintyScale::Central, UncertaintyScale::Down};
```

Comment:
Don't need to define it - use a global method `GetAllUncertaintyScales` where needed.